tools/Dashboard/DevHome.Dashboard/Services/WidgetIconService.cs

```diff
@@ -66,15 +66,8 @@ public async Task AddIconsToCacheAsync(WidgetDefinition widgetDef)
 
                 // There is a widget bug where Definition update events are being raised as added events.
                 // If we already have an icon for this key, just remove and add again in case the icons changed.
-                if (_widgetLightIconCache.ContainsKey(widgetDefId))
-                {
-                    _widgetLightIconCache.Remove(widgetDefId);
-                }
-
-                if (_widgetDarkIconCache.ContainsKey(widgetDefId))
-                {
-                    _widgetDarkIconCache.Remove(widgetDefId);
-                }
+                _widgetLightIconCache.Remove(widgetDefId);

```

Comment:
Is this safe?  Do we not need to check for existence before removing?

Reply:
`Dictionary<TKey,TValue>.Remove returns a boolean. If the key isn't found, it will continue on.